test/unit/models/index.test.ts

```diff
@@ -133,4 +133,133 @@ describe("test installation model", () => {
 			expect(subscription.jiraHost).toBe(renamedInstallPayload.baseUrl);
 		}
 	});
+
+	describe("duplicate entries in Installations table", () => {
+		it("should return no installations if jiraHost is not found", async () => {
+			const jiraHost = "https://myfakejirasite.net";
+			const installations = await Installation.getAllForHost(jiraHost);
+			const installation = await Installation.getForHost(jiraHost);
+
+			expect(installations.length).toEqual(0);
+			expect(installation).toEqual(null);
+		});
+
+		it("should return a single installation if 1 host is installed", async () => {
+			await Installation.install({
+				host: jiraHost,
+				sharedSecret: "badsecret",
+				clientKey: "1234567",
+			});
+
+			const installations = await Installation.getAllForHost(jiraHost);
+			const installation = await Installation.getForHost(jiraHost);
+
+			expect(installations.length).toEqual(1);
+			expect(installation?.jiraHost).toEqual(jiraHost);
+		});
+
+		it("should return the most recent entry if there are 2 duplicate entries", async () => {
+			// Install jira host
+			await Installation.install({
+				host: jiraHost,
+				sharedSecret: "badsecret",
+				clientKey: "1234567",
+			});
+
+			const singleInstallation = await Installation.getAllForHost(jiraHost);
+			expect(singleInstallation.length).toEqual(1);
+
+			// Install duplicate of jira host
+			await Installation.install({
+				host: jiraHost,
+				sharedSecret: "goodsecret",
+				clientKey: "12345678",
+			});
+
+			const installations = await Installation.getAllForHost(jiraHost);
+			expect(installations.length).toEqual(2);
+
+			installations.forEach((installation) => {
+				// both installations should have the same host.
+				expect(installation.jiraHost).toBe(jiraHost);
+			});
+
+			// Ids of installations should decrement to ensure most recently added is retrieved first
+			expect(installations[0].id).toBeGreaterThan(installations[1].id);
+
+			const firstRetreivedInstallation = await Installation.getForHost(jiraHost);
+			expect(firstRetreivedInstallation?.id).toEqual(installations[0].id);
```

Comment:
this is a different scenario.  Try to keep your tests to a single function that you're testing.